File: Java/414. Third Maximum Number.java
class Solution {
    public int thirdMax(int[] nums) {
        Set<Integer> three = new HashSet<>();
        
        for(int num: nums){
            three.add(num);
            
            if(three.size() > 3){
                three.remove(Collections.min(three));
            }
        }
        
        if(three.size() == 3){
            return Collections.min(three);
        }else{
            return Collections.max(three);
        }
    }
}

File: Python3/414. Third Maximum Number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        
        three = set()
        
        for num in nums:
            three.add(num)
            if len(three) > 3:
                three.remove(min(three))
                
                
        if len(three) == 3:
            return min(three)
        else:
            return max(three)
        